### PR TITLE
Update Maestro bot

### PR DESCRIPTION
**Pull request by karajas**
Resolves https://github.com/dotnet/core-eng/issues/813 
/cc @dagood

### CODE REVIEW

File: Maestro/subscriptions.json [hunk 1]
```diff
@@ -78,9 +78,9 @@
             "-Project='tests\\build.proj'",
             "--",
             "/t:UpdateDependenciesAndSubmitPullRequest",
-            "/p:GitHubUser=dotnet-bot",
-            "/p:GitHubEmail=dotnet-bot@microsoft.com",
-            "/p:GitHubAuthToken=`$(`$Secrets['DotNetBotGitHubPassword'])",
+            "/p:GitHubUser=dotnet-maestro-bot",
+            "/p:GitHubEmail=dotnet-maestro-bot@microsoft.com",
+            "/p:GitHubAuthToken=`$(`$Secrets['DotNetMaestroBotGitHubToken'])",
```
Comment (dagood) [inline Maestro/subscriptions.json]: I don't see this secret in the executor builds--are you adding it? For an overall strategy I suggest:

 1. Add `DotNetMaestroBotGitHubToken` to the VSTS defs (keeping `DotNetBotGitHubPassword` for now)
 2. Merge this PR
 3. Wait a while to make sure each def still works
 4. Remove `DotNetBotGitHubPassword` from each def

File: Maestro/subscriptions.json [hunk 2]
```diff
@@ -78,9 +78,9 @@
             "-Project='tests\\build.proj'",
             "--",
             "/t:UpdateDependenciesAndSubmitPullRequest",
-            "/p:GitHubUser=dotnet-bot",
-            "/p:GitHubEmail=dotnet-bot@microsoft.com",
-            "/p:GitHubAuthToken=`$(`$Secrets['DotNetBotGitHubPassword'])",
+            "/p:GitHubUser=dotnet-maestro-bot",
```
Comment (dagood) [inline Maestro/subscriptions.json]: dotnet-maestro-bot will need a fork already created for each repo--it doesn't create those itself, only the branches.

File: Maestro/subscriptions.json (hunk 1 above)
Comment (karajas) [inline Maestro/subscriptions.json]: Yup that was the plan. I'll add in the new token

File: Maestro/subscriptions.json (hunk 2 above)
Comment (karajas) [inline Maestro/subscriptions.json]: Ah will do.

Comment (karajas) [inline Maestro/subscriptions.json]: All the repos are now forked. 
https://github.com/dotnet-maestro-bot

File: Maestro/subscriptions.json (hunk 1 above)
Comment (karajas) [inline Maestro/subscriptions.json]: Added all the tokens to the working branches except dotnet/cli. Will file a separate issue for that.

Comment (karajas) [inline Maestro/subscriptions.json]: Filed https://github.com/dotnet/core-eng/issues/947

Comment (dagood) [inline Maestro/subscriptions.json]: I took a look at Maestro-CoreFXGeneralExecutor, and it looks like you added the variable, but didn't hook it up to the powershell script args:

    @{"DotNetBotGitHubPassword"="$(DotNetBotGitHubPassword)"}

Should be:

    @{"DotNetBotGitHubPassword"="$(DotNetBotGitHubPassword)";"DotNetBotMaestroGitHubPassword"="$(DotNetBotMaestroGitHubPassword)"}

Comment (karajas) [inline Maestro/subscriptions.json]: Updated the build definitions with the new arg

**Pull request merged**